src/composed/signed_key/key_parser_macros.rs

```diff
@@ -39,14 +39,25 @@ macro_rules! key_parser {
                 // -- One Public-Key packet
 
                 // ignore random other packets until we find something useful
-                while packets.peek().map(|p| p.tag() != $key_tag) == Some(true) {
-                    let p = packets.next().expect("peeked");
-                    warn!("ignoring unexpected packet: expected {:?}, got {:?}", $key_tag, p.tag());
+                while packets.peek().map(|p| {
+                    match p.as_ref() {
+                        Ok(p) => p.tag() != $key_tag,
+                        Err(_) => false,
+                    }
+                }) == Some(true) {
+                    match packets.next().expect("peeked") {
+                        Ok(p) => {
+                            warn!("ignoring unexpected packet: expected {:?}, got {:?}", $key_tag, p.tag());
+                            // FIXME: return error?
```

Comment:
probably should have the same logic re unsupported here?

Reply:
I think this `match` statement does what it should. My intention for this PR is that the low-level parser distinguishes three cases (instead of formerly just two):

1. Packets that it can parse and handle, and hand up to the meta-parser (we get those as `Ok`-wrapped packets, here)
2. Packets that the low level parser recognizes as safe, but `Unknown` will not be returned to the meta-parser (we don't see these packets here at all: the low-level parser skips over them, for us)
3. Packets that the low level parser can't explicitly recognize as "safe" will bubble up to the meta-parser as an error (we fail on these, here)

Reply:
(Before, no errors were ever bubbled up from the low-level parser to the meta-parsers, all "unknown" packets were skipped. So the third case didn't exist, and we didn't have the error branch here at all)